### ISSUE TITLE
Time optimization in non-rigid registrations

### ISSUE THREAD

**Issue by subhajitchatu** [kaspermarstal/SimpleElastix#82]
Hi,

I have used bspline based registration methods for registering two 3D dicom image stacks (60 slices) both using simpleITK and simpleElastix. But they took more than 40 minutes to register. How can we optimize time complexity? Is there any function to introduce parrelism(multi core or multi threading) in this case to reduce time  consumtion?
P.S. I have 4 core , 8gb RAM, 64 bit Ubuntu 14.04.

**Comment by PertuyF** [comment]
@subhajitchatu As I told you on another discussion, for any question related to SimpleElastiX parameters optimisation you should look at ElastiX's manual and participate to ElastiX's mailing list. As SimpleElastiX is "only" an API to ElastiX, what you're actually looking for is how to use ElastiX itself.

The very basic (and obvious) parameters to play with are the number of iterations and number of resolutions. Other optimisations will likely be case specific and could de related to the optimizer, metric, image sampling, resampler, and more to learn of and play with...

P.S: I never experienced 40min registration for two 60 slices images (if they are not insanely large in X and Y of course), but I had some hard time dealing with an antivirus software strategy that was hugely limiting ElastiX's performances. You may check that.

**Comment by subhajitchatu** [comment]
@PertuyF
I am really sorry for posting irrelevant questions for this forum. I will put it over ITK forum, as I have used simpleITK. Iteration in bspline was 100, with convergence factor 1e-5. I took the example from simpleElastix package example for python. For a single slice it works fine. After 1 hour it was able to iterate 8 steps and registration accuracy by diff image is only possible for monomodal images, for multimodal I think segmentation is a must. Which is not so easy for every image.
